ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/tenant_evpn_vxlan/leaf-tenant-router-bgp-evpn-vlan-aware-bundles.j2

```diff
@@ -21,20 +21,17 @@
 {%         if tenants[tenant].l2vlans is defined %}
 {%             for l2vlan in tenants[tenant].l2vlans | arista.avd.natural_sort if leaf.vlans is not none and l2vlan in leaf.vlans %}
 {%                 set l2vlan_int = l2vlan | int %}
-    {{ tenants[tenant].l2vlans[l2vlan].name }}:
-      tenant: {{ tenant }}
 {%                 if tenants[tenant].l2vlans[l2vlan].vni_override is defined %}
-      rd: "{{ overlay_loopback_network_summary | ipaddr('network') | ipmath(leaf.id + spine.nodes | length) }}:{{ tenants[tenant].l2vlans[l2vlan].vni_override }}"
+{%                     set vni = tenants[tenant].l2vlans[l2vlan].vni_override %}
 {%                 else %}
-      rd: "{{ overlay_loopback_network_summary | ipaddr('network') | ipmath(leaf.id + spine.nodes | length) }}:{{ tenants[tenant].mac_vrf_vni_base + l2vlan_int }}"
+{%                     set vni = tenants[tenant].mac_vrf_vni_base + l2vlan_int %}
 {%                 endif %}
+    {{ tenants[tenant].l2vlans[l2vlan].name }}:
+      tenant: {{ tenant }}
+      rd: "{{ leaf.evpn_rd_admin_subfield }}:{{ vni }}"
       route_targets:
         both:
-{%                 if tenants[tenant].l2vlans[l2vlan].vni_override is defined %}
-          - "{{ tenants[tenant].l2vlans[l2vlan].vni_override }}:{{ tenants[tenant].l2vlans[l2vlan].vni_override }}"
-{%                 else %}
-          - "{{ tenants[tenant].mac_vrf_vni_base + l2vlan }}:{{ tenants[tenant].mac_vrf_vni_base + l2vlan_int }}"
-{%                 endif %}
+          - "{{ leaf.evpn_rt_admin_subfield or vni }}:{{ vni }}"
```

Comment:
Could we just set a default value in the j2 logic for `leaf.evpn_rt_admin_subfield` with a `else` to catch all unsupported keys?

Then templates can just call the value and not focus on if it is correctly set or not

Reply:
+1 let's set the default value in the root template `evpn-fabric-l3leaf-yml.j2` and avoid the `or` operator.

Reply:
We will settle on inline "or" for now.